### PR TITLE
Remove use of auto and `with` for requests

### LINKED ISSUE #19: auto-enumeration not supported in older versions of python3
I'm running python 3.4.2 on my server, and the module update script gives me the following error when I try to run it:

```
factorio@munksgaard:/opt/factorio$ ./mod-updater.py -s /opt/factorio/data/server-settings.json -m /opt/factorio/mods --fact-path /opt/factorio/bin/x64/factorio --update
Traceback (most recent call last):
  File "./mod-updater.py", line 12, in <module>
    from enum import Enum, auto
ImportError: cannot import name 'auto'
```

Apparently, `auto` was not added until python 3.6 (https://docs.python.org/3/library/enum.html). Is there a way to fix your script so that it works with older versions as well?

### PR DESCRIPTION
**Pull request by Munksgaard**
Fixes #19 

This makes the script work in older versions of Python. I've tested with 3.4.2.

I haven't tested whether it still works with 3.7 or 3.6, so you should probably double-check that before merging.

### CODE REVIEW

File: mod_updater.py [hunk 1]
```diff
@@ -241,9 +241,9 @@ def _retrieve_mod_metadata(self, mod: str):
         """
         data = self.mods[mod]
         mod_url = self.mod_server_url + '/api/mods/' + mod + '/full'
-        with requests.get(mod_url) as req:
-            if req.status_code == 200:
-                data['metadata'] = req.json()
+        req = requests.get(mod_url)
+        if req.status_code == 200:
+            data['metadata'] = req.json()
```
Comment (clarfonthey) [inline mod_updater.py:246]: This actually changes what it's doing -- you need to wrap this in a `try: finally:` block to close the request. Ditto everywhere else.

**Comment by clarfonthey** [comment]
Considering how all versions before 3.6 are [officially EOL](https://pythoninsider.blogspot.com/2020/10/python-35-is-no-longer-supported.html), I don't think that this should be done, considering how it makes code quality worse and don't think that it's reasonable to hold future contributors to the standard of not using `with` statements. You mention in the other issue that your server has an old version of Python -- sounds to me like you need to update your server, instead of updating this script.

**Comment by Munksgaard** [comment]
That does sound reasonable indeed, thank you for considering though.

**Comment by pdemonaco** [comment]
@clarfonthey - totally agreed, thanks for reviewing. Apparently my email client decided Github notifications were spam so I'm just seeing this now.

**Pull request closed without merging**